### DISCUSSION TITLE
File name variable

### DISCUSSION THREAD

**Discussion by NikitaSakharov** [mtex-toolbox/mtex#745] [Ask Anything] [answered]
Hi everyone!

Hope you'll help me with my issue. What I would like to do is to find somewhere (saying "somewhere" I mean in some variable) the name of *.cpr ebsd-file, from which ebsd-data were imported.

As I can understand, filename is not saved anywhere after "EBSD.load(filename)" has worked. 

I need filename as I want to write the script for evaluating some grain boundaries characteristics and export results to excel. 
As I have many-many ebsd-files I'll need to analyse, I need some information in my excel files which would help me to distinguish between different ebsd files. Filename would be right choice for it.

And one more suggestion for the future improvements of MTEX. 
I often need to evaluate the same GB or grains characteristic from many ebsd files (for instance, I need to analyse how it is changed with annealing). So I need to do the following:
- open ebsd data file => run script => export data => open new ebsd data file => run script and so on
So is there any way for "Batch processing"? I mean I've imported several ebsd data files, assigned them several variable names (ebsd1, ebsd2, ..., ebsdN) and wrote one string of code which does all the same operations with all the ebsdN variables. This would be very helpful.

Sincerely, Nikita

**Comment by kilir** [comment, marked as answer]
Hi Nikita,
your variable `filename` is a string variable either of the full path or just the filename. So you can just use just a part of that string to either name your output file or write it to your output - it just depends how and what you are exporting.

With respect to something such as batch processing: Does all your ebsd data have the same coordinate system and material files (CS{}) ? If so, yes, of course you can automatically loop over any number of files in some directory
In very crude terms:
```
% general defition valid for all maps
CS = {... 
  'notIndexed',...
 crystalSymmetry('whatevergoeshere...)
  };

%%
infiles = dir('*.ctf') % list all files which end in .ctf

for i=1:length(infiles)
% create an EBSD variable containing the data
ebsd = EBSD.load(infiles(i).name,CS,'interface','ctf',...
  'convertEuler2SpatialReferenceFrame');

% do something

% export something - use infiles(i).name(1:end-4)
end
```
Cheers, Rüdiger

**Reply by NikitaSakharov** [reply]
Hi Rüdiger!

Many thanks for promt reply!

I'll try this!

Thanks again!

**Reply by NikitaSakharov** [reply]
And by the way. 
Is there an option to store 'filename' variable anywhere, when I use GUI import_wizard? 
It would be helpful as the filenames I use are rather long and the path to them rather long too. So it's not very 'user-friendly' to type it.

**Reply by kilir** [reply]
If you have the import_wizard generate a script, you'll find it written in the variable. If clicking it is still not sufficiently easy, you can generate a list of names from e.g. the content of a directory containing all your input (example above).

Cheers,
Rüdiger
